kie-wb-common-dmn/kie-wb-common-dmn-client/src/main/java/org/kie/workbench/common/dmn/client/editors/expressions/ExpressionEditorViewImpl.java

```diff
@@ -294,6 +434,79 @@ private void preventDefault(final ClickEvent event) {
         event.stopPropagation();
     }
 
+    private String getExpressionName() {
+        final HasName fallbackHasName = hasExpression instanceof HasName ? (HasName) hasExpression : HasName.NOP;
+        return hasName.orElse(fallbackHasName).getValue().getValue();
+    }
+
+    @SuppressWarnings("unchecked")
+    private String getTypeRef() {
+        QName qName = BuiltInType.UNDEFINED.asQName();
+        if (hasExpression instanceof HasVariable) {
+            qName = ((HasVariable<InformationItemPrimary>) hasExpression).getVariable().getTypeRef();
+        } else if (hasExpression.getExpression() != null && hasExpression.getExpression().asDMNModelInstrumentedBase().getParent() instanceof HasVariable) {
+            qName = ((HasVariable<InformationItemPrimary>) hasExpression.getExpression().asDMNModelInstrumentedBase().getParent()).getVariable().getTypeRef();
```

Comment:
are we sure `getVariable()` is always `!= null`?

Reply:
Maybe I'm misleading the type check here. We are checking if `getParent()` is instance of `HasVariable`. Isn't this enough to be sure that `getVariable` is always `!= null`? I will add an extra check.